i18n/en.json

```diff
@@ -2984,7 +2984,7 @@
   "leave_private_channel_modal.leave": "Yes, leave channel",
   "leave_private_channel_modal.message": "Are you sure you wish to leave the private channel {channel}? You must be re-invited in order to re-join this channel in the future.",
   "leave_private_channel_modal.title": "Leave Private Channel {channel}",
-  "leave_team_modal.desc": "You will be removed from all public and private channels.  If the team is private you will not be able to rejoin the team.  Are you sure?",
+  "leave_team_modal.desc": "**You will be removed from all {num_of_public_channels} public channels and {num_of_private_channels} private channels on this team**. If the team is private you will not be able to rejoin the team without an invitation. Are you sure?",
```

Comment:
This won't be grammatically correct when the user is a member of exactly 1 channel of a given type, but you can use the `{plural}` keyword to automatically pluralize words based on the number of public/private channels in the team. There's information on using that [here](https://formatjs.io/docs/core-concepts/icu-syntax#plural-format), but you'll want to write something like
```
{numOfPublicChannels, plural, one {channel} other {channels}}
```
This results in the word "channel" if `numOfPublicChannels` is equal to 1 or "channels" if it is any other number.